Fix TreeItem shows cell edit in the wrong column when `select_mode=Row` and TreeItem has multiple columns

aqfranco opened the pull request:
Fix #88892

Editor would not be brought up when clicking on all but the last column on a TreeItem with SelectMode=Row with multiple columns and all editable, and when the editor was being brought up when clicking on the last column, it was editing the first column.

The reason for that was that the draw_item function was setting the set_meta parameter for that TreeItem for each loop, ignoring if it already had that parameter set, and the select_single_item function was only considering the first column when dealing with a TreeItem with SelectMode=Row.

Fixed draw_item, gui_input, and edit_selected functions by setting set_meta parameter for each column.
Fixed select_single_item function by changing selected_col depending on input.

AThousandShips commented on scene/gui/tree.cpp:
```diff
@@ -3986,8 +3994,12 @@ bool Tree::edit_selected(bool p_force_edit) {
-	Rect2 rect = s->get_meta("__focus_rect");
+	Rect2 rect;
+	if (select_mode == SELECT_ROW) {
+		rect = s->get_meta("__focus_col_" + char(selected_col));
```
```suggestion
		rect = s->get_meta("__focus_col_" + itos(selected_col));
```

AThousandShips commented on scene/gui/tree.cpp:
```diff
@@ -3986,8 +3994,12 @@ bool Tree::edit_selected(bool p_force_edit) {
 	}
-
-	Rect2 rect = s->get_meta("__focus_rect");
+	Rect2 rect;
```
```suggestion

	Rect2 rect;
```

aqfranco committed:
Fix #88892: TreeItem shows cell edit in the wrong column when SelectMode=Row and TreeItem has multiple columns
Editor would not be brought up when clicking on all but the last column on a TreeItem with SelectMode=Row with
multiple columns and when the editor was being brought up when clicking on the last column, it was editing the first column

Fixed draw_item, gui_input and edit_selected functions by setting set_meta parameter for each column
Fixed select_single_item function by changing selected_col depending on input

aqfranco commented:
Thanks for the suggestions, already committed the changes.

akien-mga commented:
Thanks! And congrats for your first merged Godot contribution :tada:
